### PR TITLE
[CW-1408] ValidateDomainID available in goext.IAuth

### PR DESCRIPTION
**Pull request by zimnx**

### CODE REVIEW

File: cloud/keystone.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/cloudwan/gohan/schema"
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/domains"
```
Comment (cierpuchaw) [inline cloud/keystone.go]: How about aliasing this to `v3domains` for consistency with 2 imports below this one?

File: cloud/keystone.go [hunk 2]
```diff
@@ -243,6 +245,32 @@ func (client *keystoneV3Client) getTenant(filter func(*v3tenants.Project) bool)
 	return result, nil
 }
 
+func (client *keystoneV3Client) getDomain(filter func(domain *domains.Domain) bool) (*domains.Domain, error) {
+	opts := domains.ListOpts{}
```
Comment (cierpuchaw) [inline cloud/keystone.go]: `ListOpts` has a `Enabled` filter. Shouldn't we be interested only in enabled domains?

File: cloud/keystone_test_utils.go [hunk 3]
```diff
@@ -186,3 +186,16 @@ func getV3TenantsResponse() interface{} {
 		},
 	}
 }
+
+func getV3DomainsResponse() interface{} {
+	return map[string]interface{}{
+		"domains": []interface{}{
+			map[string]interface{}{
+				"id":          "Default",
+				"name":        "default",
```
Comment (cierpuchaw) [inline cloud/keystone_test_utils.go]: This is test data so it doesn't really matter, but in reality it's the other way around: id is "default" and name is "Default".

File: cloud/keystone.go (hunk 2 above)
Comment (zimnx) [reply cloud/keystone.go]: Good idea. Added to getTenants too.

File: cloud/keystone.go (hunk 1 above)
Comment (zimnx) [reply cloud/keystone.go]: why not

File: cloud/keystone_test_utils.go (hunk 3 above)
Comment (zimnx) [reply cloud/keystone_test_utils.go]: done

**Pull request merged**